### ISSUE TITLE
Unable to find python path

### ISSUE THREAD

**Issue by majormoses** [sshuttle/sshuttle#362]
I looked and I don't see any duplicates here though I found the report to ubuntu: https://bugs.launchpad.net/ubuntu/+source/sshuttle/+bug/1592853.

```
$ sshuttle -vr <REDACTED>:45750 10.0.0.0/16
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 3.6.8
firewall manager: ready method name nat.
IPv6 enabled: False
UDP enabled: False
DNS enabled: False
TCP redirector listening on ('127.0.0.1', 12300).
Starting client with Python version 3.6.8
c : connecting to server...
/bin/sh: 1: exec: python: not found
c : fatal: server died with error code 127
```

I have tried upgrading to the latest version
```
$ pip3 install sshuttle
Collecting sshuttle
  Downloading https://files.pythonhosted.org/packages/86/35/19622ccb89d1882d9c5f044c900f2a6754a1d262c07409e1661fc4dae5dc/sshuttle-0.78.5-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 2.7MB/s 
Installing collected packages: sshuttle
Successfully installed sshuttle-0.78.5
```

I can confirm that its in mine and roots path:
```
$ sudo -u root which python
/usr/bin/python
```

I am able to pass the path as a work around:
```
 $ sshuttle --python=/usr/bin/python3 -vr <REDACTED>:45750 10.0.0.0/16
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 3.6.8
firewall manager: ready method name nat.
IPv6 enabled: False
UDP enabled: False
DNS enabled: False
TCP redirector listening on ('127.0.0.1', 12300).
Starting client with Python version 3.6.8
c : connecting to server...
Starting server with Python version 3.6.7
 s: latency control setting = True
 s: available routes:
 s:   2/10.0.50.0/24
c : Connected.
firewall manager: setting up.
>> iptables -t nat -N sshuttle-12300
>> iptables -t nat -F sshuttle-12300
>> iptables -t nat -I OUTPUT 1 -j sshuttle-12300
>> iptables -t nat -I PREROUTING 1 -j sshuttle-12300
>> iptables -t nat -A sshuttle-12300 -j RETURN --dest 127.0.0.1/32 -p tcp
>> iptables -t nat -A sshuttle-12300 -j REDIRECT --dest 10.0.0.0/16 -p tcp --to-ports 12300 -m ttl ! --ttl 42
```

Let me know what else I can provide.

**Comment by brianmay** [comment]
Are you able to provide more debuging? `-vvv`?

sshuttle is supposed to try Python3 by default on the server, and only if that fails will it resort to Python2.

Based on some of the bug reports I have been getting this doesn't seem to be happening, but I cannot reproduce this myself.

**Comment by brianmay** [comment]
FYI the linked bug report is for an older version of sshuttle before the python3 support was implemented. Version 0.78.5 should have it.

Make sure you really are using the latest version you just downloaded, and you aren't loading an old version somewhere.

```
which sshuttle
```

**Comment by majormoses** [comment]
I am traveling but can give you more debug output when I return later this evening as the machine I am currently replying from does not have sshuttle installed.

**Comment by claudio-walser** [comment]
I would say i have the exact same problem, currently on a fresh install of fedora workstation 31

`which python
    /usr/bin/python`

`/usr/bin/python -V
    Python 3.7.5`

`sshuttle -vvv -r 192.168.43.21 10.5.0.0/24
Starting sshuttle proxy.
[local sudo] Password: 
firewall manager: Starting firewall with Python version 3.7.5
firewall manager: ready method name nat.
IPv6 enabled: False
UDP enabled: False
DNS enabled: False
User enabled: False
Binding redirector: 12300
TCP redirector listening on ('127.0.0.1', 12300).
TCP redirector listening with <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 12300)>.
Starting client with Python version 3.7.5
c : connecting to server...
c : executing: ['ssh', '192.168.43.21', '--', 'exec /bin/sh -c \'P=python3; $P -V 2>/dev/null || P=python; exec "$P" -c \'"\'"\'import sys, os; verbosity=3; sys.stdin = os.fdopen(0, "rb"); exec(compile(sys.stdin.read(1106), "assembler.py", "exec"))\'"\'"\'\'']
c :  > channel=0 cmd=PING len=7 (fullness=0)
/bin/sh: line 0: exec: python: not found
c : fatal: server died with error code 127
`

**Comment by brianmay** [comment]
What happens if you try running that command manually? The quoting is complicated, but this seems to work here:

```
ssh 192.168.43.21 -- "exec /bin/sh -c 'P=python3; \$P -V 2>/dev/null || P=python; exec \$P'"
<Ctrl-D to exit>
```

**Comment by claudio-walser** [comment]
I do not have python on this server, its a coreos machine. Thats a mandatory thing for sshuttle to run properly right? I just found out about this right now. Then ignore my comment, the original author of this issue might or might not have a different problem.

Thank you and kind regards

**Comment by brianmay** [comment]
@claudio-walser Yes, we do need Python on the server. Running sshd from a docker container might be a better option for you.

**Comment by claudio-walser** [comment]
Hi Brian
Yes sure, that is what I am doing now. The error message is very clear and i don't know why i did not came up with that finding earlier, without bother you and waste your time.

Everything works well, sshuttle really is a great tool, love it

Regards

**Comment by majormoses** [comment]
Hmm I am not seeing this anymore, I will report back if I can reproduce again.

**Comment by cw71** [comment]
Hi,  I am getting the same error running from Kali 2020.3.

root@kali:/usr/lib/python3/dist-packages/sshuttle# sshuttle -vvv -r cw@10.11.1.252:22000 0/0 --python=/usr/bin/python3
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 3.8.6
firewall manager: ready method name nat.
IPv6 enabled: None
UDP enabled: False
DNS enabled: False
User enabled: False
Binding redirector: 12300
TCP redirector listening on ('127.0.0.1', 12300).
TCP redirector listening with <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 12300)>.
Starting client with Python version 3.8.6
c : connecting to server...
c : executing: ['ssh', '-p', '22000', 'cw@10.11.1.252', '--', '\'/usr/bin/python3\' -c \'import sys, os; verbosity=3; sys.stdin = os.fdopen(0, "rb"); exec(compile(sys.stdin.read(1279), "assembler.py", "exec"))\'']
c :  > channel=0 cmd=PING len=7 (fullness=0)
cw@10.11.1.252's password: 
**bash: /usr/bin/python3: No such file or directory
c : fatal: server died with error code 127**
root@kali:/usr/lib/python3/dist-packages/sshuttle# **which python3**
**/usr/bin/python3**
root@kali:/usr/lib/python3/dist-packages/sshuttle# **/usr/bin/python3 -V
Python 3.8.6**

**Comment by brianmay** [comment]
@cw71 You do realize that the error is happening on the server? You need to have `/usr/bin/python3` on the server too.
